### ISSUE TITLE
`bowtie statistics` with no stdin input should default to retrieving a report from the published website

### ISSUE THREAD

**Issue by Julian** [bowtie-json-schema/bowtie#1206]
I.e. `bowtie statistics` should be equivalent to `http get 'https://bowtie.report/draft2020-12.json' | bowtie statistics`.

**Comment by adwait-godbole** [comment]
Hey @Julian, a minor suggestion from my side, let me know if you think this makes sense.

> I.e. bowtie statistics should be equivalent to http get 'https://bowtie.report/draft2020-12.json' | bowtie statistics.

instead of doing the above can we just give a "--dialect" flag to `bowtie statistics` instead ? and default that flag to `draft2020-12` ? this way we can pass other dialect options as well!

**Comment by Julian** [comment]
I considered that and I forget why I discarded it -- but yeah sure try that and let's see if we run into any issue!

**Comment by adwait-godbole** [comment]
@Julian we could also potentially take this further that if the user is just calling `bowtie statistics` then we could show the stats for all the dialects at once ?

**Comment by Julian** [comment]
Sounds good as well!

**Comment by adwait-godbole** [comment]
Also I see that you've just set `default="-"` on the report type because of which on running just `bowtie statistics` nothing shows up on CLI until we manually `Ctrl + C` and then an error pops up saying "report-not-json". Instead can't we just prompt the user if he passes nothing to just pass something ?

**Comment by Julian** [comment]
You can use the same "convention" we use elsewhere there and default to `-` only when it's not a tty (and then your pexpect-style test).

**Comment by adwait-godbole** [comment]
Okay got it! One more thing for testing this should we make real http calls to `"https://bowtie.report/draft2020-12.json"` ? If not then I am not sure how does the mocking part work exactly ? Reminds me of the same GitHub API mocking we wanted to do perhaps.

**Comment by Julian** [comment]
No, never real HTTP calls in tests -- you can probably leave out testing the HTTP portion for now, as you say done correctly it's similar to the GitHub API story but also not something terribly critical for the moment to test, we can have some "live" tests by simply running the command in GitHub actions at some point perhaps but it's quite straightforward functionality so I'm not rushing yet to do the mocking work.
